apps/src/templates/EnhancedSafeMarkdown.jsx

```diff
@@ -87,10 +87,14 @@ export class ClickableTextWrapper extends React.Component {
 
   renderClickableText(node) {
     const clickableTextAll = node.querySelectorAll('b.clickable-text');
-    for (const clickableText of clickableTextAll) {
+    clickableTextAll.forEach((clickableText, index) => {
       const id = clickableText.dataset.id;
+      const extraClass = ` clickable-text-${index}`;
+      if (!clickableText.className.includes(extraClass)) {
+        clickableText.className += extraClass;
+      }
```

Comment:
significant whitespace always feels icky to me.

First off, is this always valid? I'm not sure what guarantees we have about the list of classnames, but since this is expecting the classname to _always_ have a space in front of it, is there a case where we could only have a classname we're looking for at the start of the string? i.e., we have `className = 'foo'` and we look to see if ` foo` appears in the string and fail as a result?

As a non-blocking suggestion, what about using a set? Something like this:

```
const clickableTextClassNames = new Set(clickableText.className.split(/\s+/));
clickableTextAll.forEach( (clickableText, index) => {
  const id = clickableText.dataset.id;
  const className = `clickable-text-${index}`; // leading space not required
  clickableTextClassNames.add(className)
  clickableText.onclick = () => this.props.handleInstructionsTextClick(id)
})

clickableText.className = Array.from(clickableTextClassNames).sort().join(' ');
```

Reply:
Agreed that this is not great, and more generally it's unfortunate that we have to protect against adding the classname repeatedly at all.  The new code is clever but I find significantly more complex than the proposed code, and the proposed code does seem to work reliably, since we are detecting first that it has the class `.clickable-text`, and the addition of ` clickable-text-${index}` does seem to always be applied just once.